### PR TITLE
chore: put tighther deadlines on test runtime

### PR DESCRIPTION
**Pull request by dpc**
Will rebase after #5176 lands

This is to prevent silent regressions.

I wish we could go lower. Most devimint-based tests are already below 40s, and unit-test like tests should be OK under 20s, IMO.

We'll get there.

### CODE REVIEW

**Review by bradleystachurski — APPROVED** [review]

**Comment by bradleystachurski** [comment]
```
TIMEOUT [  60.007s] fedimint-wallet-tests::fedimint_wallet_tests peg_out_fail_refund
```

Perhaps a bit too tight?

**Comment by bradleystachurski** [comment]
Timing out now with the updated `parallel` 130s timeout

```
       > 36    :       1714530498.663     130.344      0       28      -1      15      run_test_for_versions bckn_gw_not_client FM: current CLI: current GW: current
       >
       > 63      :       1714530587.481     130.960      0       28      -1      15      run_test_for_versions bckn_gw_not_client FM: current CLI: current GW: current
```

Perhaps we can resolve that one by breaking up the gateway tests into smaller chunks?

**Comment by dpc** [comment]
> Perhaps we can resolve that one by breaking up the gateway tests into smaller chunks?

I already did and split it in two.

It's OK. I'll increase to 150.

**Review by bradleystachurski — DISMISSED** [review]

**Comment by maan2003** [comment]
IMO time limit should be defined per test

**Comment by dpc** [comment]
> IMO time limit should be defined per test

Could be done as an additional one, but seems like somewhat manual thing to maintain.

**Comment by dpc** [comment]
:crossed_fingers: This change is cursed and keeps triggering new unseen yet flakes.

**Review by bradleystachurski — DISMISSED** [review]

**Comment by bradleystachurski** [comment]
> This change is cursed

I'm becoming concerned that if we land this, subsequent PRs will have to get lucky to pass the merge queue. Should we hold off?

**Comment by dpc** [comment]
> I'm becoming concerned that if we land this, subsequent PRs will have to get lucky to pass the merge queue. Should we hold off?

If we're hitting tests legitimately taking more time the deadlines we should raise them. But if it's unrelated flakes (which seems to have happened a lot here), holding off doesn't help.

**Comment by dpc** [comment]
Last failure might be fixed by #5179 so will re-add after it lands.

**Review by bradleystachurski — APPROVED** [review]

**Pull request merged**
